components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java

```diff
@@ -1437,6 +1438,25 @@ EnvironmentPropertiesDTO getEnvironmentSpecificAPIProperties(String apiUuid, Str
      */
     Environment getEnvironment(String organization, String uuid) throws APIManagementException;
 
+    /**
+     * Get Endpoint details by providing API UUID.
+     *
+     * @param uuid      Unique Identifier of API
+     * @return List<APIEndpointInfo> Object
+     * @throws APIManagementException if an error occurs while retrieving endpoints details
+     */
+    List<APIEndpointInfo> getAllAPIEndpointsByUUID(String uuid) throws APIManagementException;
+
+    /**
+     * Get Endpoint details by providing API UUID.
```

Comment:
method description is the same as above method. Lets put a meaning full method description here.

Reply:
Also mention why APIUUID is needed to here? Basically why we cant just use the endpointUUID only. (Because endpointUUID is same for endpoints created for different revisions of same API)

Reply:
done

Reply:
can u update the method description meaningfully, why API uuid is needed